omero/sysadmins/unix/server-centos7.txt

```diff
@@ -0,0 +1,178 @@
+OMERO.server installation on CentOS 7
+=====================================
+
+This is an example walkthrough for installing OMERO on CentOS 7, using
+a dedicated system user, and should be read in conjunction with
+:doc:`install-web`. You can use this as a guide
+for setting up your own test server. For production use you should also read
+the pages listed under :ref:`index-optimizing-server`.
+
+Example shell scripts for installing OMERO are included in-line, and can be
+downloaded.
+These instructions assume your Linux distribution is configured with a UTF-8
+locale (this is normally the default).
+
+For convenience in this walkthrough the main OMERO configuration options have
+been defined as environment variables. When following this walkthrough you can
+either use your own values, or alternatively source the following file:
+
+.. literalinclude:: walkthrough/settings.env
+   :start-after: Substitute
+
+:download:`settings.env <walkthrough/settings.env>`
+
+
+Prerequisites
+-------------
+
+**The following steps are run as root.**
+
+
+Install Ice, Java, PostgreSQL:
+
+.. literalinclude:: walkthrough/step01_centos7_deps.sh
+
+| :download:`step01_centos7_deps.sh <walkthrough/step01_centos7_deps.sh>`
+
+Create an omero system user, and a directory for the OMERO repository:
+
+.. literalinclude:: walkthrough/step02_all_setup.sh
+
+:download:`step02_all_setup.sh <walkthrough/step02_all_setup.sh>`
+
+
+Create a database user and initialize a new database for OMERO:
+
+.. literalinclude:: walkthrough/step03_all_postgres.sh
+
+:download:`step03_all_postgres.sh <walkthrough/step03_all_postgres.sh>`
+
+
+Installing OMERO.server
+-----------------------
+
+**The following steps are run as the omero system user.**
+
+Download, unzip and configure OMERO, and create a
+configuration file for Nginx. The rest of this walkthrough assumes the
+OMERO.server is installed into the home directory of the omero system user.
+
+Note that this script requires the same environment variables that were set
+earlier in `settings.env`, so you may need to copy and/or source this file as
+the omero user.
+
+.. literalinclude:: walkthrough/step04_all_omero.sh
+
+Installing a web server
+-----------------------
+
+To Deploy OMERO.web, you can use either Nginx or Apache.
+Follow the steps to install your chosen web server.
+
+Nginx
+^^^^^
+
+**The following steps are run as root.**
+
+Install Nginx, copy the Nginx OMERO configuration file into the
+Nginx configuration directory, and disable the default configuration:
+
+.. literalinclude:: walkthrough/step05_centos7_nginx.sh
+
+| :download:`step05_centos7_nginx.sh <walkthrough/step05_centos7_nginx.sh>`
+
+Apache
+^^^^^^
+
+**The following steps are run as root unless specified.**
```

Comment:
Sorry for last minute spotting but I thin this is a bit confusing. Could we move line 93 ``As the **omero**...`` up to 95 ``walkthrough/setup_omero_apache24...``  before installing apache then all the commands needs to be run as root. This will much two scripts down below.